x-pack/plugins/alerting/common/routes/rule/response/schemas/v1.ts

```diff
@@ -37,156 +47,397 @@ export const notifyWhenSchema = schema.oneOf(
+          schema.literal(ruleExecutionStatusErrorReasonV1.VALIDATE),
+          schema.literal(ruleExecutionStatusWarningReasonV1.MAX_EXECUTABLE_ACTIONS),
+          schema.literal(ruleExecutionStatusWarningReasonV1.MAX_ALERTS),
+          schema.literal(ruleExecutionStatusWarningReasonV1.MAX_QUEUED_ACTIONS),
+        ],
+        {
+          meta: {
+            description: 'Warning of last rule execution.',
+          },
+        }
+      )
+    )
+  ),
+  outcome_msg: schema.maybe(
+    schema.nullable(
+      schema.arrayOf(
+        schema.string({
+          meta: {
+            description: 'Outcome message generated during last rule run.',
+          },
+        })
+      )
     )
   ),
-  outcome_msg: schema.maybe(schema.nullable(schema.arrayOf(schema.string()))),
   alerts_count: schema.object({
-    active: schema.maybe(schema.nullable(schema.number())),
-    new: schema.maybe(schema.nullable(schema.number())),
-    recovered: schema.maybe(schema.nullable(schema.number())),
-    ignored: schema.maybe(schema.nullable(schema.number())),
+    active: schema.maybe(
+      schema.nullable(
+        schema.number({
+          meta: {
+            description: 'Number of active alerts during last run.',
+          },
+        })
+      )
+    ),
+    new: schema.maybe(
+      schema.nullable(
+        schema.number({
+          meta: {
+            description: 'Number of new alerts during last run.',
+          },
+        })
+      )
+    ),
+    recovered: schema.maybe(
+      schema.nullable(
+        schema.number({
+          meta: {
+            description: 'Number of recovered alerts during last run.',
+          },
+        })
+      )
+    ),
+    ignored: schema.maybe(
+      schema.nullable(
+        schema.number({
+          meta: {
+            description: 'Number of ignore alerts during last run.',
+          },
+        })
+      )
+    ),
   }),
 });
 
-export const monitoringSchema = schema.object({
-  run: schema.object({
-    history: schema.arrayOf(
-      schema.object({
-        success: schema.boolean(),
-        timestamp: schema.number(),
-        duration: schema.maybe(schema.number()),
-        outcome: schema.maybe(ruleLastRunSchema),
-      })
+export const monitoringSchema = schema.object(
+  {
+    run: schema.object(
+      {
+        history: schema.arrayOf(
+          schema.object({
+            success: schema.boolean({
+              meta: { description: 'Indicates whether the rule run was success or not.' },
+            }),
+            timestamp: schema.number({ meta: { description: 'Time of rule run.' } }),
+            duration: schema.maybe(
+              schema.number({ meta: { description: 'Duration of the rule run.' } })
+            ),
+            outcome: schema.maybe(ruleLastRunSchema),
+          }),
+          { meta: { description: 'History of the rule run.' } }
+        ),
+        calculated_metrics: schema.object({
```

Comment:
Couldn't find any docs, comments or example which could help explain this metrics. Would be great if someone can suggest description for `calculated_metrics ` object.

Reply:
How about `"Calculation of different percentiles and success ratio"`?